rdkcentral/Lightning-UI-Components
# fix: update global typography props when new theme is set

## Description
Prior to this change, when switching between different themes, the objects representing the styles for text (ultimately passed to the Lightning Text texture) would merge values from the previous and new text style object. For example: ThemeA has fontStyle property on a text style object, ThemeB does not have that property on the same text style object. Swapping from ThemeA to ThemeB would result in a combined text style object between the 2 themes (the correct text styles from ThemeB in addition to the fontStyle property from ThemeA).

This changes how text style properties are parsed by withThemeStyles to ensure a new object is generated for text styles when a theme changes. The new object will contain all default values of the Lightning Text texture as well as those specified as style properties in the component.


## References
LUI-874


## Testing
1. add a new theme that contains text styles with additional fields that are not in the base theme's text style objects
2. navigate to a component that utilizes a text style that changes between the themes
3. switch between themes
ER: The component should only have text styles from the selected theme applied.


## Automation



## Checklist
